pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -147,7 +147,7 @@ def optimizer_step(
         """Hook to run the optimizer step."""
         if isinstance(model, pl.LightningModule):
             closure = partial(self._wrap_closure, model, optimizer, optimizer_idx, closure)
-        optimizer.step(closure=closure, **kwargs)
+        optimizer.step(closure=closure, **kwargs)  # type: ignore[call-arg]
```

Comment:
I've always found passing `**kwargs` here weird. I guess it's useful for weird Optimizer implementations which violate the Liskov Substitution Principle.

One way we could get around it is doing

```python
    @runtime_checkable
    class _Optimizer(Protocol):
        def step(self, closure: Callable[[], Any], **kwargs: Any) -> Any:
            ...

    def optimizer_step(
        self,
        model: Union["pl.LightningModule", Module],
        optimizer: _Optimizer,
        optimizer_idx: int,
        closure: Callable[[], Any],
        **kwargs: Any,
    ) -> None:
```

Or maybe I'm just too excited about having learned a new thing and want to use it where I shouldn't.

Do you have any thoughts about this @ananthsub?

Reply:
I think it is fine for us to type: ignore because we know this is about custom Optimizers and their step method. I can add a comment if you want. 

If you insist, please go ahead and commit it directly since you know best

Reply:
It was more of a side comment. This does not block this PR